pkg/sources/github/github.go

```diff
@@ -1165,8 +1165,25 @@ func extractGistID(urlParts []string) string {
 	return urlParts[len(urlParts)-1]
 }
 
+// isGistUrl returns true if the URL path is of a gist
 func isGistUrl(urlParts []string) bool {
-	return strings.EqualFold(urlParts[0], "gist.github.com") || (len(urlParts) == 4 && strings.EqualFold(urlParts[1], "gist"))
```

Comment:
I removed the `len(urlParts) == 4` condition since it was overly restrictive and failed to match enterprise gist URLs with different path structures.